crates/router/src/core/payments/helpers.rs

```
@@ -1474,6 +1479,111 @@ pub fn validate_customer_information(
     }
 }
 
+#[cfg(feature = "v1")]
+pub async fn validate_card_ip_blocking_for_business_profile(
+    state: &SessionState,
+    request: &api_models::payments::PaymentsRequest,
+    fingerprnt: masking::Secret<String>,
+    card_testing_guard_config: &diesel_models::business_profile::CardTestingGuardConfig,
+) -> RouterResult<String> {
+    let mut ip: IpAddr = IpAddr::V4(Ipv4Addr::UNSPECIFIED);
+
+    if let Some(browser_info) = &request.browser_info {
+        if let Ok(browser_info_parsed) =
+            serde_json::from_value::<payments::BrowserInformation>(browser_info.clone())
+        {
+            if let Some(browserip) = browser_info_parsed.ip_address {
+                ip = browserip;
+            }
+        }
+    }
+
+    let cache_key = format!(
+        "{}_{}_{}",
+        consts::CARD_IP_BLOCKING_CACHE_KEY_PREFIX,
+        fingerprnt.peek(),
+        ip
+    );
+
+    let unsuccessful_payment_threshold = card_testing_guard_config.card_ip_blocking_threshold;
+
+    match services::card_testing_guard::get_blocked_count_from_cache(state, &cache_key).await {
+        Ok(Some(unsuccessful_payment_count)) => {
+            if unsuccessful_payment_count >= unsuccessful_payment_threshold {
+                Err(errors::ApiErrorResponse::PreconditionFailed {
+                    message: "Blocked due to suspicious activity".to_string(),
+                })?
+            } else {
+                Ok(cache_key)
+            }
+        }
+        Ok(None) => Ok(cache_key),
+        Err(_) => Err(errors::ApiErrorResponse::InternalServerError)?,
```

Avoid discarding the error, the error should be lifted into `internalservererror`.

---

I have resolved it, thanks